projects/laji/src/app/shared/service/collection.service.ts

```diff
@@ -61,4 +97,35 @@ export class CollectionService extends AbstractCachedHttpService<ICollectionRang
       switchMap(cols => hasMore ? this.fetchWarehouseCollections(page + 1, cols) : of(cols))
     );
   }
+
+  getCollectionsTree(): Observable<any> {
+    return this.graphQlService.query({
+      query: this.TREE_QUERY,
+      errorPolicy: 'all',
+      fetchPolicy: 'cache-first',
+      context: {
+        headers: {
+          'x-timeout': '180000'
+        }
+      }
+
+    }).pipe(
+      map(({data}) => data),
+    );
+  }
+
+  getCollectionsAggregate(page = 1, collections = []): Observable<any> {
+    let hasMore = false;
+    return this.warehouseApi.warehouseQueryAggregateGet({cache: true}, ['document.collectionId'], undefined, 1000, page).pipe(
+      tap(data => hasMore = data.lastPage && data.lastPage > page),
+      map(data => data.results || []),
+      map(data => data.map(d => {
+        return {
+          id: IdService.getId(d?.aggregateBy?.['document.collectionId']),
+          count: d.count
+        };
+      })),
+      map(cols => [...collections, ...cols]),
+      switchMap(cols => hasMore ? this.fetchWarehouseCollections(page + 1, cols) : of(cols)));
```

Comment:
should this be `this.getCollectionsAggregate` instead of `this.fetchWarehouseCollections`?